### PR TITLE
Fix `BA2027.EnableSourceLink` cause `ExceptionLoadingPdb`

### PR DESCRIPTION
**Pull request by shaopeng-gh**
Fix `BA2027.EnableSourceLink` unexpectedly cause `ExceptionLoadingPdb` error when the PDB file is missing.

Please see details in https://github.com/microsoft/binskim/issues/986

The cause is this PR [[WIP] Add BA2027.EnableSourceLink](https://github.com/microsoft/binskim/pull/657)
missed setting `LogPdbLoadException` => false. (Pattern found from other rules)
I used the get history version to locate this exact PR changed the behavior.

Fix: add it, and add tests to prevent future new rules missing the same property set.

Additionally, I have question about other existing rules that have this `LogPdbLoadException` as true.
Are they bugs? What condition we want this true / false? This is a question about original design when we introduce this flag. 
Let me know I can create a bug if need fixes.

### CODE REVIEW

File: ReleaseHistory.md [hunk 1]
```diff
@@ -18,6 +18,7 @@
 * DEP: Update `Sarif.Sdk` submodule from [bc8cb57 to fd6e615](https://github.com/microsoft/sarif-sdk/compare/bc8cb57...fd6e615). Reference [SARIF SDK Release History](https://github.com/microsoft/sarif-sdk/blob/fd6e615/ReleaseHistory.md).
 * NEW: Add `--disable-telemetry` argument to disable telemetry collection.
 * BUG: Fix `ERR998.ExceptionInAnalyze`: `InvalidOperationException: Unrecognized crypto HRESULT: 0x80096011` for check `BA2022.SignSecurely` when the signature is malformed, by adding missing error code to error description mappings. [969](https://github.com/microsoft/binskim/pull/969)
+* BUG: Fix `BA2027.EnableSourceLink` unexpectedly cause `ExceptionLoadingPdb` error when the PDB file is missing. [988](https://github.com/microsoft/binskim/pull/988)
```
Comment (stacywray) [inline ReleaseHistory.md:21]: - Change "cause" to "causes"
- Add a period to the end of the line after the 988 link. The BUG above is also missing a period at the end after 969.

**Review by stacywray — CHANGES_REQUESTED** [review]
Requesting a few minor changes:
- Change "cause" to "causes".
- Add period at end of line. The BUG immediately above also needs a period at the end of the line.

**Review by stacywray — APPROVED** [review]
Looks good.

File: src/Test.UnitTests.BinSkim.Rules/RulePropertyTests.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+using FluentAssertions;
+
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.IL.Rules
+{
+    public class RulePropertyTests
+    {
+        private static readonly string[] ExpectedLogPdbLoadExceptionRules = new string[]
```
Comment (shaopeng-gh) [inline src/Test.UnitTests.BinSkim.Rules/RulePropertyTests.cs:18]: Please see PR desc.

File: src/Test.UnitTests.BinaryParsers/TestData/PE/Determinism/E_PDB_NOT_FOUND/Managed_x64_VS2022_CSharp_Net70_Default.dll
Comment (shaopeng-gh) [inline src/Test.UnitTests.BinaryParsers/TestData/PE/Determinism/E_PDB_NOT_FOUND/Managed_x64_VS2022_CSharp_Net70_Default.dll:1]: The old test was based on BA2027 throw that error, now that is fixed, so need to change to another file triggered by other rules.

**Pull request merged**